src/h2olog/h2olog.cc

```diff
@@ -125,6 +126,42 @@ static void show_process(pid_t pid)
     infof("Attaching pid=%d (%s)", pid, cmdline);
 }
 
+static void drop_root_privilege(void)
+{
+    if (getuid() == 0) {
+        const char *sudo_gid = getenv("SUDO_GID");
+        if (sudo_gid == NULL) {
+            fprintf(stderr, "Error: failed to read the SUDO_GID env variable\n");
```

Comment:
Should we better say that the environment variable is not set, rather than "failed to read"?

Reply:
Great catch! Addressed in: b6c5d3f